src/battlegoals/partyBattleGoals.tsx

```diff
@@ -2,78 +2,177 @@ import * as React from "react";
-}
-
-function drawDistinctBattleGoals(allBattleGoals: Array<number>, count:number): Array<number> {
+function drawDistinctBattleGoals(allBattleGoals: Array<BattleGoal>, count: number): Array<BattleGoal> {
   return shuffle(allBattleGoals).slice(0, count);
 }
 
-enum BattleGoalFlavour {
-  Vanilla = 'Vanilla',
-  Community = 'Community',
+interface Picks {
+  [picks: number]: number
 }
 
 interface PartyBattleGoalsState {
-  flavour: BattleGoalFlavour;
+  includeVanilla: boolean;
+  includeCommunity: boolean;
+  drawnBattleGoals: BattleGoal[];
+  currentPlayer: void | number;
+  picks: Picks;
 }
 
 export default class PartyBattleGoals extends React.Component<NoProps, PartyBattleGoalsState> {
+  private storage: Storage = window.localStorage;
 
   constructor(props: NoProps) {
     super(props);
-    this.handleFlavourChange = this.handleFlavourChange.bind(this);
+    this.handleDrawBattleGoals = this.handleDrawBattleGoals.bind(this);
+    this.toggleVanilla = this.toggleVanilla.bind(this);
+    this.toggleCommunity = this.toggleCommunity.bind(this);
+    this.logState = this.logState.bind(this);
+    this.handlePlayerToggle = this.handlePlayerToggle.bind(this);
+    this.handlePlayerPick = this.handlePlayerPick.bind(this);
     this.state = {
-      flavour: BattleGoalFlavour.Vanilla
+      includeVanilla: false,
+      includeCommunity: false,
+      drawnBattleGoals: [],
+      currentPlayer: undefined,
+      picks: {}
     }
   }
 
-  private handleFlavourChange(event: React.ChangeEvent<HTMLSelectElement>) {
-    this.setState({ flavour: event.target.value as BattleGoalFlavour }, () => {
-      console.log(this.state)
-    });
+  componentDidMount() {
+    const dtoAsString = this.storage.getItem('partyBattleGoalState');
+    if (dtoAsString !== null) {
+      const dto = JSON.parse(dtoAsString);
+      const stateFromStore = {
+        includeVanilla: dto.includeVanilla,
+        includeCommunity: dto.includeCommunity,
+        drawnBattleGoals: dto.battleGoalIds.map(battleGoalByGlobalId),
+        picks: dto.picks || {}
+      };
+      this.setState(stateFromStore);
+    }
   }
 
-  public render(): React.ReactNode {
-    const vanilla = this.state.flavour === BattleGoalFlavour.Vanilla;
-
-    const allBattleGoals = vanilla ? battleGoals() : communityBattleGoals();
-    const battleGoalsPerPlayer = partition(2, drawDistinctBattleGoals(allBattleGoals, 8));
-    const containerStyle = {
-      'display': 'flex',
-      'flex-direction': 'row'
-    };
-    const playerBattleGoalsStyle = {
-      'display': 'flex',
-      'flex-direction': 'column',
-      'padding': '0 0.25em 0 '
+  componentWillUnmount(): void {
+    const dto = {
+      includeCommunity: this.state.includeCommunity,
+      includeVanilla: this.state.includeVanilla,
+      battleGoalIds: this.state.drawnBattleGoals.map(it => it.globalCardId),
+      picks: this.state.picks
     };
+    this.storage.setItem('partyBattleGoalState', JSON.stringify(dto));
```

Comment:
Can you rename this to e.g. storeState() and call whenever the state changes? Otherwise I fear the "save" logic might be too brittle (e.g. w.r.t. to user closing the tab). Just replace the calls to "logState" with "storeState" :)

Reply:
done

Reply:
# 👻 👍